rules/prefer-replace-all.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const quoteString = require('./utils/quote-string');
+
+function isRegexWithGlobalFlag(node) {
+	const {type, regex} = node;
+
+	return type === 'Literal' && regex && regex.flags === 'g';
+}
+
+function isLiteralCharactersOnly(node) {
+	const searchPattern = node.regex.pattern;
+	const specialCharacters = searchPattern.match(/[$()*+.?[\\\]^{}]/g);
+	const specialCharactersCount = specialCharacters ? specialCharacters.length : 0;
+	const escapedSpecialCharacters = searchPattern.match(/\\[$()*+.?[\\\]^{}]/g);
+	const escapedSpecialCharactersCount = escapedSpecialCharacters ? escapedSpecialCharacters.length : 0;
+	return specialCharactersCount === 2 * escapedSpecialCharactersCount;
+}
```

Comment:
I'm sure this can be simplified, can you check it?

Reply:
Hmm. One idea would be 
```js
function isLiteralCharactersOnly(node) { 
   const searchPattern = node.regex.pattern;
   const specialCharactersCount = searchPattern.split(/[$()*+.?[\\\]^{}]/).length - 1;
   const escapedSpecialCahractersCount = searchPattern.split(/\\[$()*+.?[\\\]^{}]/).length - 1;
   return specialCharactersCount === 2 * escapedSpecialCharactersCount; 
}
```
It could then turn into:
```js
function isLiteralCharactersOnly(node) { 
   const searchPattern = node.regex.pattern;
   return  searchPattern.split(/[$()*+.?[\\\]^{}]/).length - 1 === 2 *(searchPattern.split(/\\[$()*+.?[\\\]^{}]/).length - 1)
}
```

And if we want, could even turn it into:
```js
function isLiteralCharactersOnly(node) { 
   return  node.regex.pattern.split(/[$()*+.?[\\\]^{}]/).length - 1 === 2 *(node.regex.pattern.split(/\\[$()*+.?[\\\]^{}]/).length - 1)
}
```

I don't know which is most readable though.

Reply:
If I understand correctly, this function means check specialCharacters(like `.`, including `\`) length (A) and escaped specialCharacters like `\.` or `\\` length (B), and A should be B * 2 ?

Reply:
Correct. Every special character is escaped.

Reply:
If so, how about 

```js
function isLiteralCharactersOnly(node) { 
   return  /[$()*+.?[\\\]^{}]/.test(pattern.replace(/\\[$()*+.?[\\\]^{}]/g, '')
}
```

equal?

On cellphone, not sure

Reply:
Very clever. Yeah, that should work. I'll try it out when I'm able.

Reply:
function isLiteralCharactersOnly(node) { 
   return  !/[$()*+.?[\\\]^{}]/.test(pattern.replace(/\\[$()*+.?[\\\]^{}]/g, '')
}

That worked but when I merged in your other changes, tests started failing

Reply:
@jmoore914 see

https://github.com/sindresorhus/eslint-plugin-unicorn/pull/488#discussion_r362688499